From 2a1b5db1b6701074061c4aa4212494cbf5feadb3 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 16 Jan 2018 17:52:49 +0800
Subject: [PATCH 3203/5242] MLK-17385: dma: imx-sdma: update sdma script for
 multi fifo on SAI

commit  a8547abdc16fdccc55bcd64bb5959998f6d007e6 from
https://source.codeaurora.org/external/imx/linux-imx.git

update sdma script for multi fifo SAI on i.mx8MQ.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/imx-sdma.c                     |    4 ++--
 include/linux/platform_data/dma-imx-sdma.h |    2 ++
 include/linux/platform_data/dma-imx.h      |    2 ++
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index 5e52d21..4669e6b 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -9,7 +9,7 @@
 // Based on code from Freescale:
 //
 // Copyright 2004-2016 Freescale Semiconductor, Inc. All Rights Reserved.
-// Copyright 2017 NXP.
+// Copyright 2018 NXP.
 
 #include <linux/init.h>
 #include <linux/iopoll.h>
@@ -1918,7 +1918,7 @@ static void sdma_issue_pending(struct dma_chan *chan)
 #define SDMA_SCRIPT_ADDRS_ARRAY_SIZE_V1	34
 #define SDMA_SCRIPT_ADDRS_ARRAY_SIZE_V2	38
 #define SDMA_SCRIPT_ADDRS_ARRAY_SIZE_V3	41
-#define SDMA_SCRIPT_ADDRS_ARRAY_SIZE_V4	42
+#define SDMA_SCRIPT_ADDRS_ARRAY_SIZE_V4	44
 
 static void sdma_add_scripts(struct sdma_engine *sdma,
 		const struct sdma_script_start_addrs *addr)
diff --git a/include/linux/platform_data/dma-imx-sdma.h b/include/linux/platform_data/dma-imx-sdma.h
index 30e676b..e257209 100644
--- a/include/linux/platform_data/dma-imx-sdma.h
+++ b/include/linux/platform_data/dma-imx-sdma.h
@@ -54,6 +54,8 @@ struct sdma_script_start_addrs {
 	s32 mcu_2_ecspi_addr;
 	/* End of v3 array */
 	s32 mcu_2_zqspi_addr;
+	s32 mcu_2_sai_addr;
+	s32 sai_2_mcu_addr;
 	/* End of v4 array */
 };
 
diff --git a/include/linux/platform_data/dma-imx.h b/include/linux/platform_data/dma-imx.h
index 18ed10c..f149a67 100644
--- a/include/linux/platform_data/dma-imx.h
+++ b/include/linux/platform_data/dma-imx.h
@@ -1,5 +1,6 @@
 /*
  * Copyright 2004-2015 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2018 NXP.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
@@ -43,6 +44,7 @@ enum sdma_peripheral_type {
 	IMX_DMATYPE_ASRC_SP,	/* Shared ASRC */
 	IMX_DMATYPE_SAI,	/* SAI */
 	IMX_DMATYPE_HDMI,	/* HDMI Audio */
+	IMX_DMATYPE_MULTI_SAI,	/* MULTI FIFOs For Audio */
 };
 
 enum imx_dma_prio {
-- 
1.7.9.5

